modules/caddyhttp/reverseproxy/healthchecks.go

```diff
@@ -377,7 +380,7 @@ func (h *Handler) doActiveHealthCheck(dialInfo DialInfo, hostAddr string, networ
 	ctx = context.WithValue(ctx, caddyhttp.VarsCtxKey, map[string]any{
 		dialInfoVarKey: dialInfo,
 	})
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	req, err := http.NewRequestWithContext(ctx, h.HealthChecks.Active.Method, u.String(), nil)
```

Comment:
If `Method` is not set by the user, the value here will be empty string.

Reply:
Yep. But apparently the std lib treats that as GET: https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/request.go;l=882

Still, that's not obvious, nor documented I think, so maybe we should set it explicitly if it's empty in the config.

Reply:
> If `Method` is not set by the user, the value here will be empty string.

Indeed, and if `http.NewRequestWithContext` is called with the empty string it defaults to `GET`.

Reply:
> Yep. But apparently the std lib treats that as GET: https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/request.go;l=883-887
> 
> Still, that's not obvious, nor documented I think, so maybe we should set it explicitly if it's empty in the config.

Fair enough.

Reply:
I've added a check and set to solve the concern.